### PR TITLE
TINKERPOP-1633 use official jbcrypt 0.4 in maven central

### PR DESCRIPTION
**Pull request by dysmento**
targeting tp31 branch for this version bump of jbcrypt to 0.4

### CODE REVIEW

**Comment by spmallette** [comment]
I've had some time to review this. I have a few additional tweaks if you don't mind. In the pom.xml you edited you can removed this `<repository>`:

```xml
<repository>
  <id>jitpack.io</id>
  <url>https://jitpack.io</url>
</repository>
```

You can also remove this entry in the docs:

https://github.com/apache/tinkerpop/blame/2decc8132ff8726e35e722b8b35475dfa4b2b47c/docs/src/reference/gremlin-applications.asciidoc#L53

There is an entry in `grapeConfig.xml` in the docker directories and a second entry on the gremlin-applications.asciidoc, but both look like those changes were added on the tp32 branch and not here on tp31 - I will take care of that stuff after merging your PR.

**Comment by dysmento** [comment]
all set.

**Comment by spmallette** [comment]
All tests pass with `docker/build.sh -t -n -i`

VOTE +1

**Comment by robertdale** [comment]
transferring my vote from #556
VOTE +1

File: docs/src/reference/gremlin-applications.asciidoc [hunk 1]
```diff
@@ -50,7 +50,6 @@ the defaults.  TinkerPop recommends the following configuration in that file:
       </filesystem>
       <ibiblio name="codehaus" root="http://repository.codehaus.org/" m2compatible="true"/>
```
Comment (pluradj) [inline docs/src/reference/gremlin-applications.asciidoc]: We can remove `codehaus` also. It went out of service a couple years ago.

Comment (spmallette) [inline docs/src/reference/gremlin-applications.asciidoc]: true - i can do that as part of the merge though

**Pull request merged**